photutils/psf/building/catalogs.py

```diff
@@ -0,0 +1,1537 @@
+        self._iter_fit_status = iter_fit_status
+
+    @property
+    def iter_fit_eps(self):
+        """
+        Set/Get PSF fit error status. The value of `None` indicates that
+        the fit has not been performed and the value of 0 indicates that the
+        fit was successful.
+
+        """
+        return self._iter_fit_eps
+
+    @iter_fit_eps.setter
+    def iter_fit_eps(self, iter_fit_eps):
+        self._iter_fit_eps = iter_fit_eps
+
+    @property
+    def ignore(self):
+        """
+        Set/Get 'ignore' attribute that indicates whether this star
+        should be used when constructing a PSF (`True` - use this star,
+        `False` - ignore this star).
+
+        """
+        return self._ignore
+
+    @ignore.setter
+    def ignore(self, ignore):
+        self._ignore = ignore
+
+    @property
+    def pixel_scale(self):
+        """
+        Set/Get pixel scale (in arbitrary units). Pixel scale of the star
+        can be used in conjunction with pixel scale of a PSF to determine
+        PSF's oversampling factor. Either a single floating point value or
+        a 1D iterable of length at least 2 (x-scale, y-scale) can be provided.
+        When getting pixel scale, a tuple of two values is returned with a
+        pixel scale for each axis.
+
+        """
+        return self._pixscale
+
+    @pixel_scale.setter
+    def pixel_scale(self, pixel_scale):
+        if pixel_scale == 'wcs':
+            if wcs:
+                pixel_scale = wcs.utils.proj_plane_pixel_scales(wcs)
+
+            else:
+                raise ValueError("'wcs' attribute not set.")
+
+        if hasattr(pixel_scale, '__iter__'):
+            if len(pixel_scale) != 2:
+                raise TypeError("Parameter 'pixel_scale' must be either a "
+                                "scalar or an iterable with two elements.")
+            self._pixscale = (float(pixel_scale[0]), float(pixel_scale[1]))
+
+        else:
+            self._pixscale = (float(pixel_scale), float(pixel_scale))
+
+    @property
+    def peak_fit_box(self):
+        """ Set/Get ``peak_fit_box`` property. See `Star` for more details. """
+        return self._peak_fit_box
+
+    @peak_fit_box.setter
+    def peak_fit_box(self, peak_fit_box):
+        self._peak_fit_box = peak_fit_box
+
+    @property
+    def peak_search_box(self):
+        """
+        Set/Get ``peak_search_box`` property. See `Star` for more details.
+
+        """
+        return self._peak_search_box
+
+    @peak_search_box.setter
+    def peak_search_box(self, peak_search_box):
+        self._peak_search_box = peak_search_box
+
+
+def expand_starlist(stars):
+    """ Get a list of all stars including "hidden" linked stars. """
+
+    if not hasattr(stars, '__iter__'):
+        stars = [stars]
+
+    expanded_starlist = []
+    for s in stars:
+        expanded_starlist += s.get_linked_list()
+
+    return expanded_starlist
+
+
+def extract_stars(images, catalogs, common_catalog=None, extract_size=11,
```

Comment:
I do not know the functionality of photutil's finder. Make sure it can "link" somehow the same star in the sky to its image in different images. This is required in order to impose "constaints" (as discussed in the previous sprint).